### PR TITLE
KIALI-1343 Service details metrics: show only inbound

### PR DESCRIPTION
**Pull request by jotak**
Simplifies code for state & charts definitions

JIRA: https://issues.jboss.org/browse/KIALI-1343

Screenshot:
screenshot

### CODE REVIEW

**Comment by lucasponce** [comment]
+1
I think that in Workload Details page and Application Details page we will maintain both Inbound and Outboud metrics. Isn't it ?

**Comment by lucasponce** [comment]
I guess this doesn't need any additional work from backend. right ?

File: src/pages/ServiceDetails/ServiceMetrics.tsx [hunk 1]
```diff
@@ -114,19 +103,17 @@ class ServiceMetrics extends React.Component<ServiceMetricsProps, ServiceMetrics
...
+          Object.keys(prevState.charts).forEach(k => {
+            const chart = prevState.charts[k];
+            const histo = metrics.histograms[k];
+            if (histo) {
+              chart.metrics = histo;
+            } else {
+              chart.metrics = metrics.metrics[k];
+            }
+          });
+          return prevState;
```
Comment (israel-hdez) [inline src/pages/ServiceDetails/ServiceMetrics.tsx]: IIRC, `prevState` should be used as a constant. Here you are changing some values.
I tried the code and it works. But not sure if React may do weird things under some circumstances.

Comment (josejulio) [reply src/pages/ServiceDetails/ServiceMetrics.tsx]: You are right @israel-hdez 
from the [docs](https://reactjs.org/docs/react-component.html#setstate):
> prevState is a reference to the previous state. It should not be directly mutated. Instead, changes should be represented by building a new object based on the input from prevState
> ...
> The output of the updater is shallowly merged with prevState

Comment (jotak) [reply src/pages/ServiceDetails/ServiceMetrics.tsx]: ok I'll fix that

**Comment by jotak** [comment]
@lucasponce :
> I think that in Workload Details page and Application Details page we will maintain both Inbound and Outboud metrics. Isn't it ?
Yes, and as we mentioned previously, maybe it would be nice then to show them in separate tabs

> I guess this doesn't need any additional work from backend. right ?
Correct

**Pull request merged**